libdnf/repo/solvable/Dependency.cpp

```diff
@@ -0,0 +1,157 @@
+#include "Dependency.hpp"
+#include "libdnf/utils/utils.hpp"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+#   include <solv/pool_parserpmrichdep.h>
+#ifdef __cplusplus
+}
+#endif
+
+
+#define NUMBER_OF_ITEMS_IN_REL_DEPENDENCY 3
+
+class Dependency::Impl {
+public:
+    Impl();
+    Impl(const Impl &source);
+    ~Impl() = default;
+private:
+    friend class Dependency;
+
+    void parseAndCreateDependency(const std::string &dependency);
+    void createRelationalDependency(const std::vector<std::string> &results);
+    bool isRichDependency(const std::string &dependency) const;
+    int determineComparisonType(const std::string &type);
+    Pool *getPool();
+
+    Id libsolvableId;
+    DnfSack *sack;
+    int type;
+};
+
+Dependency::Impl::Impl()
+        : libsolvableId(-1)
```

Comment:
What about to move Id, and DnfSack into private part of Dependency? By that way you spare 1 pointer. If there will be sack and Id (pointer, int), hide it behind another pointer is quite "funny".

Reply:
You mean I should remove `class Impl`?

Reply:
If so I won't do it. Read about `Pimpl idiom` and its pros and cons. But I think you should know it, you used it yourself.